lib/gprdperm.gi

```diff
@@ -750,7 +762,35 @@ InstallMethod( ImagesRepresentative,
         [ IsEmbeddingImprimitiveWreathProductPermGroup,
           IsMultiplicativeElementWithInverse ], 0,
     function( emb, g )
-    return g ^ WreathProductInfo( Range( emb ) ).perms[ emb!.component ];
+    local info, degI, x, shift, domI, degG, i, k, l;
+    info := WreathProductInfo(Range(emb));
+    degI := info.degI;
+    if emb!.component = degI + 1 then
```

Comment:
I would suggest to reverse the logic: instead of having a loooong case followed by a one line case, how about writing it more like this:
```suggestion
    if emb!.component <> degI + 1 then
      return g ^ info.perms[emb!.component];
    fi;
```
followed by the long case, which now can be indented one level less.

Reply:
Yes, I will change this.